app/lib/api_support/forbid_params.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ApiSupport
+  class UnpermittedParameters < StandardError
+    def initialize(unpermitted_keys)
+      msg = "Unpermitted parameters: #{unpermitted_keys}"
+      super msg
+    end
+  end
+
+  module ForbidParams
+    extend ::ActiveSupport::Concern
+    include Params
+
+    included do
+      class_attribute :_forbid_params_action
+      class_attribute :_forbid_params_whitelist, default: []
+
+      rescue_from UnpermittedParameters do |error|
+        render_error error.message, status: :unprocessable_entity
+      end
+    end
+
+    class_methods do
+      def forbid_extra_params(action, options = {})
+        before_action :unpermitted_parameters_check
+        self._forbid_params_action = action
+
+        return unless options[:whitelist]
+
+        self._forbid_params_whitelist = [*options[:whitelist]]
+      end
+    end
+
+    private
+
+    def forbid_params_whitelist
```

Comment:
all these methods, if they are only internal methods, not supposed to be called by anything outside this module, maybe rename them to `_***`?

If they are to be called by code outside this module, then this is alright.

Reply:
They are already marked as private methods, why the prefix? I don't see the prefix in any other private method in porta.

Reply:
To avoid conflicts. I'm always very paranoid about modules. As they can interfere with each other :(
Maybe longer, more specific/clear names?

Reply:
It's OK, I renamed then.